File: library/src/main/java/com/chuckerteam/chucker/internal/ui/HomePageAdapter.java
package com.chuckerteam.chucker.internal.ui;

import android.content.Context;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentStatePagerAdapter;

import com.chuckerteam.chucker.R;
import com.chuckerteam.chucker.internal.ui.error.ErrorListFragment;
import com.chuckerteam.chucker.internal.ui.traffic.TrafficFragment;

import java.lang.ref.WeakReference;

/**
 * @author Olivier Perez
 */
class HomePageAdapter extends FragmentStatePagerAdapter {

    static final int SCREEN_HTTP_INDEX = 0;
    static final int SCREEN_ERROR_INDEX = 1;

    private final WeakReference<Context> context;

    HomePageAdapter(Context context, FragmentManager fragmentManager) {
        super(fragmentManager);
        this.context = new WeakReference<>(context);
    }

    @NonNull
    @Override
    public Fragment getItem(int position) {
        if (position == SCREEN_HTTP_INDEX) {
            return new TrafficFragment();
        } else {
            return ErrorListFragment.newInstance();
        }
    }

    @Override
    public int getCount() {
        return 2;
    }

    @Override
    public CharSequence getPageTitle(int position) {
        Context context = this.context.get();
        if (context == null) {
            return null;
        }
        if (position == SCREEN_HTTP_INDEX) {
            return context.getString(R.string.chucker_tab_network);
        } else {
            return context.getString(R.string.chucker_tab_errors);
        }
    }
}

File: library/src/main/java/com/chuckerteam/chucker/internal/ui/HomePageAdapter.kt
package com.chuckerteam.chucker.internal.ui

import android.content.Context
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentStatePagerAdapter
import com.chuckerteam.chucker.R
import com.chuckerteam.chucker.internal.ui.throwable.ThrowableListFragment
import com.chuckerteam.chucker.internal.ui.transaction.TransactionListFragment
import java.lang.ref.WeakReference

internal class HomePageAdapter(context: Context, fragmentManager: FragmentManager) :
    FragmentStatePagerAdapter(fragmentManager, BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT) {
    private val context: WeakReference<Context> = WeakReference(context)

    override fun getItem(position: Int): Fragment = if (position == SCREEN_HTTP_INDEX) {
        TransactionListFragment.newInstance()
    } else {
        ThrowableListFragment.newInstance()
    }

    override fun getCount(): Int = 2

    override fun getPageTitle(position: Int): CharSequence? =
        context.get()?.getString(
            if (position == SCREEN_HTTP_INDEX) {
                R.string.chucker_tab_network
            } else {
                R.string.chucker_tab_errors
            }
        )

    companion object {
        const val SCREEN_HTTP_INDEX = 0
        const val SCREEN_THROWABLE_INDEX = 1
    }
}
